### PR TITLE
#3682: processing get-action.triggered events

### LINKED ISSUE #3682: Clean-up remediation-service to not control the remediation sequence
**Current status:** In its current implementation, the remediation-service controls the execution of the remediation sequence by sending out the `action.triggered` event and waiting for `action.finished` and `evaluation.finished`. Besides, waiting for 10 minutes is hard-coded into the services. 

Should we call the task: `get-action` ? Since we have the same discussion here: https://github.com/keptn/spec/issues/57 --> `get-action`

**Clean-up remediation-service:** The remediation-service does not control the sequence, but:
* it subscribes to a `get-action.triggered` event,
* it analysis the event payload and determines the next action to execute by looking into the `remediation.yaml`.
* If the `remediation.yaml` defines multiple actions for one problem and the one was already be executed, the remediation-service must be smart enough to take the next one in the list of actions. 

**Clean-up of spec:**
- By implementing this approach, the reserved Keptn Task `remediation` is not needed anymore: https://github.com/keptn/spec/blob/master/shipyard.md#remediation

As a result of this clean-up task, the shipyard-controller takes over the responsibility for sending `action.triggered`, and `evaluation.triggered` - since this is modeled in the shipyard. 

**Design decisions:**
* The [sh.keptn.event.problem.open](https://github.com/keptn/spec/blob/master/cloudevents.md#problem) event does not work as trigger for the `remediation` sequence --> it will be the `sh.keptn.event.remediation.triggered` that kicks off the sequence. 
* Can the issue https://github.com/keptn/spec/issues/59 be resolved then?

**Definition of Done**: 
- [ ] Adaptation of integration tests
- [ ] The remediation sequence execution is explicitly modeled by the shipyard, e.g.:
```
- name: "production" 
  sequences: 
  - name: remediation
      tasks:
      - name: get-action
      - name: action
      - name: evaluation
        triggeredAfter: "10m"
```

### PR DESCRIPTION
**Pull request by warber**
closes #3682 

According to the issue, with this PR the behavior of the remediation service was changed to only process `sh.keptn.event.get-action` events

### CODE REVIEW

**Comment by warber** [comment]
@check-spelling-bot apply changes

**Comment by warber** [comment]
@check-spelling-bot apply [changes](https://api.github.com/repos/keptn/keptn/issues/comments/832510526)

**Comment by warber** [comment]
> @check-spelling-bot apply [changes](https://api.github.com/repos/keptn/keptn/issues/comments/832510526).

**Comment by warber** [comment]
> @check-spelling-bot apply [changes](https://api.github.com/repos/keptn/keptn/issues/comments/832510526).

**Comment by warber** [comment]
> @check-spelling-bot apply [changes](https://api.github.com/repos/keptn/keptn/issues/comments/832728474).

**Comment by warber** [comment]
We should merge this PR as soon all related stories are ready:

https://github.com/keptn-contrib/dynatrace-service/issues/287
https://github.com/keptn/keptn/issues/3683
https://github.com/keptn/keptn/issues/3698
https://github.com/keptn/keptn/issues/3755

File: remediation-service/handler/remediation.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	return remediationData, nil
+}
+
+func GetNextAction(remediation *v0_1_4.Remediation, problemDetails v0_2_0.ProblemDetails, actionIndex int) (*v0_2_0.ActionInfo, error) {
+	rootCause := problemDetails.RootCause
+	problemTitle := problemDetails.ProblemTitle
+
+	var actions []v0_1_4.RemediationActionsOnOpen
+	for _, r := range remediation.Spec.Remediations {
+		if r.ProblemType == rootCause || r.ProblemType == problemTitle {
```
Comment (bacherfl) [inline remediation-service/handler/remediation.go:30]: would it make sense to first go through all remediations and check if a remediation for the root cause is available?
This way, having a remediation that matches the problem title would override a more specific one, if it is defined earlier in the shipyard. e.g.:

```
problemTitle = "response time degradation"
rootCause = "failure rate increase"

remediations:
  - problemType: "response time degradation"
    actionsOnOpen:
      - name: "some action"
      
  - problemType: "failure rate increase"
    actionsOnOpen:
      - name: "other action that would be better suited"
```

In this example, the service would determine that the correct action to take would be the first one (based on the problem title), while the other one that addresses the root cause will not be returned.

**Review by bacherfl — CHANGES_REQUESTED** [review]

File: remediation-service/handler/remediation.go (hunk 1 above)
Comment (warber) [reply remediation-service/handler/remediation.go:30]: good catch, will do it

**Pull request closed without merging**